core/io/resource_uid.cpp

```diff
@@ -109,6 +109,13 @@ ResourceUID::ID ResourceUID::text_to_id(const String &p_text) const {
 }
 
 ResourceUID::ID ResourceUID::create_id() {
+	// mbedTLS may not be fully initialized when the ResourceUID is created, so we
+	// need to lazily instantiate the random number generator.
+	if (crypto == nullptr) {
+		crypto = memnew(CryptoCore::RandomGenerator);
+		((CryptoCore::RandomGenerator *)crypto)->init();
+	}
+
```

Comment:
This is very hacky, but we can't call `init` during the constructor, because mbedTLS won't be fully setup yet (the threading functions are set in module initialization CORE which happens after the creation of the ResourceUID singleton).

I've been thinking on dropping the `CryptoCore` usage completely in this class, and instead use `OS::get_singleton()->get_entropy()`.
This might have some negative impact (as it polls the OS entropy generator every time, instead of just using it to initialize a PRNG).
At the same time, given the amount of entropy bytes is low, and the function is mostly called from the editor it might be a worthy change just to clean it up a bit.

Reply:
Do we really need cryptographic PRNG to generate UIDs used for resources? Wouldn't PCG be "good enough" and let us avoid this hack?